java/src/jmri/jmrit/withrottle/UserInterface.java

```diff
@@ -416,18 +416,16 @@ public void serviceQueued(ZeroConfServiceEvent se) {
     public void servicePublished(ZeroConfServiceEvent se) {
         try {
             try {
-                InetAddress addr = se.getAddress();
-                // most addresses are Inet6Address objects, 
-                if (addr instanceof Inet4Address && !addr.isLoopbackAddress()) {
+                InetAddress addr = se.getDNS().getInetAddress();
+                //output last good ipV4 address to the window to support manual entry
+                if (addr instanceof Inet4Address) {
```

Comment:
On Feb 11, 2016, at 8:10 PM, Randall Wood <notifications@github.com> wrote:
> In java/src/jmri/jmrit/withrottle/UserInterface.java:
> 
> > @@ -416,18 +416,16 @@ public void serviceQueued(ZeroConfServiceEvent se) {
> >      public void servicePublished(ZeroConfServiceEvent se) {
> >          try {
> >              try {
> > -                InetAddress addr = se.getAddress();
> > -                // most addresses are Inet6Address objects, 
> > -                if (addr instanceof Inet4Address && !addr.isLoopbackAddress()) {
> > +                InetAddress addr = se.getDNS().getInetAddress();
> > +                //output last good ipV4 address to the window to support manual entry
> > +                if (addr instanceof Inet4Address) {
> Is there a reason not to disallow the loopback address here?
> 
Testing, if nothing else.  It is certainly valid for an emulated device to use a loop back address.

Paul